### ISSUE TITLE
Multiplex mode - user consent issues.

### ISSUE THREAD

**Issue by VINISHVV** [Ylianst/MeshCentral#3631]
In my current setup admin user can access the any system without user consent. For rest of the users user consent is mandatory.

I noticed that if a normal user try to connect the system first and wait for user approval. At the same time if admin tried to connect same system then system waits for user consent approval for admi  too.

**Comment by JSuenram** [comment]
Hi, I guess at the moment this is by design. In Multiplex mode the first user creates the "stream" and all subsequent connections only grab that stream directly from the MC-Server. So the 2nd connection does not really reach the Agent but only captures the current connection which already exists on the server. I believe this can be fixed, but for the moment the only workaround is to connect admin first or disable Multiplex.

**Comment by Ylianst** [comment]
Right. The desktop multiplexor makes it so the MeshAgent will see only one session, so it does cause some issues like this. I am not sure how to work around this.

**Comment by JSuenram** [comment]
> Right. The desktop multiplexor makes it so the MeshAgent will see only one session, so it does cause some issues like this. I am not sure how to work around this.

Maybe check if existing connection has consent enabled and compare if new session would be allowed connect without consent and bypass Multiplex in this situation? Or merge the session later? Just an idea, duno if that is possible...
